data/lua/wml/kill.lua

```diff
@@ -50,19 +50,25 @@ function wesnoth.wml_actions.kill(cfg)
 			-- In other words, the attacker's weapon. The attacker is the secondary unit.
 			-- In the victory animation, this is simply swapped.
 			if primary then
+				local found_attack = nil
 				if secondary_unit then
-					primary = secondary_unit:find_attack(primary)
-				else
+					found_attack = secondary_unit:find_attack(primary)
+				end
+				if not found_attack then
 					primary = wesnoth.units.create_weapon(primary)
```

Comment:
`primary` gets assigned on this line, but that's unconditionally overwritten by the following `primary = found_attack`.

Reply:
While the code now works, I think the conditional blocks should be swapped so it's `if found_attack ... else` rather than `if not found_attack ... else`.